includes/Checker/CLI_Runner.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\CLI_Runner
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+/**
+ * CLI Runner class.
+ *
+ * @since n.e.x.t
+ */
+class CLI_Runner extends Abstract_Check_Runner {
+
+	/**
+	 * Checks if the current request is a CLI request for the Plugin Checker.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool
+	 */
+	public function is_plugin_check() {
+		if ( empty( $_SERVER['argv'] ) ) {
+			return false;
+		}
+
+		if (
+			'wp' === $_SERVER['argv'][0] &&
+			'plugin' === $_SERVER['argv'][1] &&
+			'check' === $_SERVER['argv'][2]
```

Comment:
These array accesses could cause PHP warnings. We either need to have an `isset` for each of them, or (probably simpler) we need to bail before if `count( $_SERVER['argv'] ) < 3 )`.